word_racer/session.py

```diff
@@ -0,0 +1,295 @@
+                if self.bonus[(y,x)] == 3:
+                    col = TRIPLE_COLOR
+            if self.board[x][y] != "#":
+                if y != 5 and self.board[x][y+1] != "#":
+                    d.line([64*x+32,64*y+32,64*x+32,64*y+96], fill=LINE_COLOR, width=1)
+                if y != 5 and x != 5 and self.board[x+1][y+1] != "#":
+                    d.line([64*x+32,64*y+32,64*x+96,64*y+96], fill=LINE_COLOR, width=1)
+                if x != 5 and self.board[x+1][y] != "#":
+                    d.line([64*x+32,64*y+32,64*x+96,64*y+32], fill=LINE_COLOR, width=1)
+                if y != 0 and x != 5 and self.board[x+1][y-1] != "#":
+                    d.line([64*x+32,64*y+32,64*x+96,64*y-32], fill=LINE_COLOR, width=1)
+                d.ellipse([64*x+16,64*y+16,64*x+48,64*y+48], fill=col)
+                txt = self.board[x][y][0].upper()+self.board[x][y][1:]
+                w, h = d.textsize(txt, font=f)
+                d.text((64*x+32-w/2, 64*y+32-h/2), txt, fill=LINE_COLOR, font=f)
+        img.save(str(self.output_image_path), "PNG")
+
+    def _gen_board(self):
+        def generate_letter(level):
+            prob = np.array(_FREQ[level])
+            prob = prob.astype(float)/sum(prob)
+            l = chr(ord('a')+np.random.choice(26,p=prob))
+            if l == "q":
+                l = "qu"
+            return l
+
+        levelmap = _LEVELS[self.level]
+        self.bonus = _BONUSES[self.level]
+        for i in range(36):
+            if levelmap[i] == "#":
+                self.board[i//6][i%6] = "#"
+            else:
+                self.board[i//6][i%6] = generate_letter(self.level)
+
+    def _solve_init(self):
+        # Return generator of words found
+        alphabet = ''.join(set(''.join([''.join(i) for i in self.board])))
+        bogglable = re.compile('[' + alphabet + ']{3,}$', re.I).match
+
+        words = set(word.lower().rstrip('\n') for word in open(str(self.dictDir)) if bogglable(word))
+        prefixes = set(word[:i] for word in words
+                    for i in range(2, len(word)+1))
+
+        def solve():
+            for y, row in enumerate(self.board):
+                for x, letter in enumerate(row):
+                    for result in extending(letter, ((x, y),)):
+                        yield result
+
+        def extending(prefix, path):
+            if prefix in words:
+                yield (prefix, path)
+            for (nx, ny) in neighbors(path[-1][0], path[-1][1]):
+                if (nx, ny) not in path:
+                    prefix1 = prefix + self.board[ny][nx]
+                    if prefix1 in prefixes:
+                        for result in extending(prefix1, path + ((nx, ny),)):
+                            yield result
+
+        def neighbors(x, y):
+            for nx in range(max(0, x-1), min(x+2, self.ncols)):
+                for ny in range(max(0, y-1), min(y+2, self.nrows)):
+                    yield (nx, ny)
+
+        return solve()
```

Comment:
Have you tested this algorithm for run time and correctness?

Reply:
I've played the game on my local instance. It generally runs in ~1-2 seconds (which is already overshadowed by the 3s delay that I added. I think the time between rounds is quite fast.)

I also took this code from StackExchange lol (https://stackoverflow.com/questions/746082/how-to-find-list-of-possible-words-from-a-letter-matrix-boggle-solver#750012)